### PR TITLE
Fix bargap 0 bug

### PR DESCRIPTION
**Pull request by etpinard**
fixes https://github.com/plotly/plotly.js/issues/708

Bar traces with `layout.bargap: 0` and unset `marker.line.width` have been broken since https://github.com/plotly/plotly.js/pull/124 back in `v1.3.0` (oof) when the `Bar` trace module got split up into separate files. 

More specifically, when this [line](https://github.com/plotly/plotly.js/pull/124/files#diff-e8fbe22d44eec250fb191d681f452b56L69) got moved to this [line](https://github.com/plotly/plotly.js/pull/124/files#diff-da932f44998140058d62ade6a4074073R33) **without** its default overriding argument made the effective line width in the plot step a `NaN` (see [here](https://github.com/plotly/plotly.js/blob/e246331c0bd34e74807aa406e86d3b1b91c5f45a/src/traces/bar/plot.js#L69)).

This PR fixes this bug by correctly setting the `marker.line.width` default value to `0` in the bar attributes (note that the scatter `marker.line.width` attribute does not have a `dflt` value in its attribute s as it depends on whether or not the trace has bubbles - see [here](https://github.com/plotly/plotly.js/blob/e246331c0bd34e74807aa406e86d3b1b91c5f45a/src/traces/scatter/marker_defaults.js#L54))

### CODE REVIEW

File: src/traces/bar/attributes.js [hunk 1]
```diff
@@ -15,6 +15,19 @@ var extendFlat = require('../../lib/extend').extendFlat;
 var scatterMarkerAttrs = scatterAttrs.marker;
 var scatterMarkerLineAttrs = scatterMarkerAttrs.line;
 
+var markerLineWidth = extendFlat({},
+    scatterMarkerLineAttrs.width, { dflt: 0 });
```
Comment (etpinard) [inline src/traces/bar/attributes.js]: here it is!

File: src/traces/histogram/attributes.js [hunk 2]
```diff
@@ -124,14 +119,7 @@ module.exports = {
     },
     ybins: makeBinsAttr('y'),
 
-    marker: extendFlat({}, {
-        showscale: barMarkerAttrs.showscale,
-        line: extendFlat({},
-            {width: extendFlat({}, barMarkerLineAttrs.width, {dflt: 0})},
```
Comment (etpinard) [inline src/traces/histogram/attributes.js]: The histogram attribute had `dflt: 0` even before this PR, explaining why this bug didn't extend to histogram traces.

File: src/traces/histogram/attributes.js [hunk 3]
```diff
@@ -124,14 +119,7 @@ module.exports = {
...
-            colorAttributes('marker.line'))
-    },
-        colorAttributes('marker')
-    ),
+    marker: barAttrs.marker,
```
Comment (etpinard) [inline src/traces/histogram/attributes.js]: Cleaner, right?

**Comment by mdtusz** [comment]
Good to get this finally fixed. 💃

**Pull request merged**